src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LargeLongHopscotchSet.java

```diff
@@ -0,0 +1,224 @@
+        //Truncate numElements to largest int possible
+        final int idealPartitionSize = LongHopscotchSet.getPartitionSize(numElements);
+        //Number of elements per partition
+        final int elementsPerPartitionGuess = Math.min(idealPartitionSize, maxPartitionSize);
+        final int partitions = (int) Math.ceil(((double) numElements) / elementsPerPartitionGuess);
+        final int elementsPerPartition = (int)Math.ceil(numElements / (double)partitions);
+
+        for (int i = 0; i < partitions; i++) {
+            sets.add(new LongHopscotchSet(elementsPerPartition));
+        }
+        numSets = sets.size();
+    }
+
+    protected void serialize(final Kryo kryo, final Output stream) {
+        final boolean oldReferences = kryo.getReferences();
+        kryo.setReferences(false);
+
+        stream.writeInt(sets.size());
+        for (final LongHopscotchSet set : sets) {
+            kryo.writeObject(stream, set);
+        }
+
+        kryo.setReferences(oldReferences);
+    }
+
+    public boolean add(final long entryValue) {
+        final int setIndex = setIndexOf(entryValue);
+        final LongHopscotchSet setRef = sets.get(setIndex);
+        return setRef.add(entryValue);
+    }
+
+    public void addAll(final long[] entryValues) {
+        for (final long val : entryValues) {
+            add(val);
+        }
+    }
+
+    public long size() {
+        long sum = 0;
+        for (final LongHopscotchSet s : sets) {
+            sum += s.size();
+        }
+        return sum;
+    }
+
+    public Collection<LongHopscotchSet> getSets() {
+        return sets;
+    }
+
+    public long capacity() {
+        long sum = 0;
+        for (final LongHopscotchSet s : sets) {
+            sum += s.capacity();
+        }
+        return sum;
+    }
+
+    public boolean contains(final long key) {
+        return sets.get(setIndexOf(key)).contains(key);
+    }
+
+    public boolean containsAll(final long[] vals) {
+        for (final long val : vals) {
+            if (!contains(val))
+                return false;
+        }
+        return true;
+    }
+
+    public LongIterator iterator() {
+        return new LargeLongHopscotchSetIterator();
+
+    }
+
+    public boolean remove(final long key) {
+        return sets.get(setIndexOf(key)).remove(key);
+    }
+
+    public boolean removeAll(final long[] set) {
+        boolean result = false;
+        for (final long val : set) {
+            if (remove(val)) result = true;
+        }
+        return result;
+    }
+
+    public boolean isEmpty() {
+        return this.size() == 0;
+    }
+
+    private int setIndexOf(final long val) {
+        return Integer.remainderUnsigned(longHash(val), numSets);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
See comments in LongHopscotchSet about making this work with any QueryableLongSet.